service/hub/auth/authn.go

```diff
@@ -0,0 +1,45 @@
+package auth
+
+import (
+	"context"
+	"log"
+
+	"github.com/labring/sealos/pkg/client-go/kubernetes"
+	"github.com/labring/service/hub/api"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	log.Println("authn plugin init function called")
+}
+
+type SealosAuthn struct {
+	api.Authenticator
+}
+
+func (a SealosAuthn) Authenticate(user string, password api.PasswordString) (bool, api.Labels, error) {
+	log.Println("Authenticate for user:", user)
+	client, err := kubernetes.NewKubernetesClientByConfigString(string(password))
+	if err != nil {
+		return false, nil, err
+	}
+	nodes, err := client.Kubernetes().CoreV1().Nodes().List(context.Background(), v1.ListOptions{})
```

Comment:
sealos cloud user kubeconfig don't have the permission to list nodes

Reply:
Only for coding test. Later I will implement this function in its entirety.